packages/utils/src/stacktrace.ts

```diff
@@ -14,7 +14,10 @@ export function createStackParser(...parsers: StackLineParser[]): StackParser {
 
   return (stack: string, skipFirst: number = 0): StackFrame[] => {
     const frames: StackFrame[] = [];
-    for (const line of stack.split('\n').slice(skipFirst)) {
+    const lines = stack.split('\n');
+
+    for (let i = skipFirst; i < lines.length; i++) {
```

Comment:
it seems like we have a stacktrace limit of 50 frames, however we still end up parsing the entire stacktrace - I added a return at the end of each iteration that will exit once frames array exceeds the max size